packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -415,24 +415,26 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
         },
       );
 
-      // Register the ability to quickly mark elements as dirty.
-      // The performance of this method may be improved with additional
-      // information from https://github.com/flutter/flutter/issues/46195.
       registerServiceExtension(
         name: 'fastReassemble',
         callback: (Map<String, Object> params) async {
-          final String className = params['class'] as String;
+          final FastReassemblePredicate fastReassemblePredicate = _debugFastReassembleMethod;
+          if (fastReassemblePredicate == null) {
+            throw FlutterError('debugFastReassembleMethod must be set to use fastReassemble.');
+          }
           void markElementsDirty(Element element) {
             if (element == null) {
               return;
             }
-            if (element.widget?.runtimeType?.toString()?.startsWith(className) ?? false) {
+            if (fastReassemblePredicate(element.widget)) {
               element.markNeedsBuild();
             }
             element.visitChildElements(markElementsDirty);
           }
           markElementsDirty(renderViewElement);
-          return <String, String>{'Success': 'true'};
+          await endOfFrame;
+          _debugFastReassembleMethod = null;
```

Comment:
Move this up to right after you've assigned it to fastReassemblePredicate?

Reply:
Done